File: files/CH-082.py
import pandas as pd

path = "CH-082 Attendance Date.xlsx"
input = pd.read_excel(path, usecols="B:D", skiprows=1, nrows = 5)
test = pd.read_excel(path, usecols="F:G", skiprows=1)

result = input.assign(Date=input.apply(lambda row: pd.date_range(row['From'], row['To'], freq='D'), axis=1)) \
    .explode('Date') \
    .groupby('Date') \
    .agg({'Supervisor': lambda x: ', '.join(x)}) \
    .reset_index()
result.rename(columns={'Supervisor': 'Supervisors'}, inplace=True)

print(result.equals(test)) # True   

File: files/000-099/082/CH-082.R
library(tidyverse)
library(readxl)

path = "files/CH-082 Attendance Date.xlsx"
input = read_excel(path, range = "B2:D7")
test  = read_excel(path, range = "F2:G14")

result = input %>%
  mutate(Date = map2(From, To, seq, by = "days")) %>%
  unnest(Date) %>%
  summarise(Supervisors = str_c(Supervisor, collapse = ", "), .by = "Date") 

identical(result, test)
#> [1] TRUE